edx_lint/cmd/amnesty.py

```diff
@@ -92,7 +92,7 @@ def fix_pylint(line, errors):
     if current:
         yield PYLINT_EXCEPTION_REGEX.sub(disable_string, line)
     else:
-        yield re.sub(r'($\s*)', disable_string + r'\1', line)
+        yield re.sub(r'($\s*)', disable_string + r'\1', line, count=1)
```

Comment:
You said that `\n` is treated differently now? I don't understand this change. Do you have a link so I can understand what changed?

Reply:
here's the issue [https://bugs.python.org/issue32308](url)

Apparently re.sub used to fail to match the empty match after adjacent match.  so in python<=3.6
```
>>> hello = 'hello\n'
>>> disable_str = '###'
>>> re.sub(r'($\s*)', disable_str + r'\1', x)
'hello###\n'
```
It only matches the `\n` character.
in python3.7, it also matches the empty "line" after the `\n` like so:
```
>>> re.sub(r'($\s*)', disable_str + r'\1', x)
'hello###\n###'
``` 
Since all the lines have the trailing newline it was doubling every pylint amnesty comment.  
Unfortunately, changing `*` to `+` won't work because of the case [where there's a missing newline at end of file](https://github.com/edx/edx-lint/blob/e9830f370bb333dd975fb325f3b47da55fbc06f3/test/test_amnesty.py#L312)